src/Uno.UI/UI/Xaml/Controls/MenuFlyout/CascadingMenuHelper.cs

```diff
@@ -0,0 +1,976 @@
+
+						foreach (var element in elements)
+						{
+							if (ownerAsUIE == element || subMenuPresenterAsUIE == element)
+							{
+								isOwnerOrSubMenuHit = true;
+								break;
+							}
+						}
+
+						if (!isOwnerOrSubMenuHit)
+						{
+							elements = VisualTreeHelper.FindElementsInHostCoordinates(point, subMenuPresenterAsUIE, true /* includeAllElements */);
+
+							foreach (var element in elements)
+							{
+								if (ownerAsUIE == element || subMenuPresenterAsUIE == element)
+								{
+									isOwnerOrSubMenuHit = true;
+									break;
+								}
+							}
+						}
+
+						// To close the sub menu, the pointer must be outside of the opened chain of sub-menus.
+						if (!isOwnerOrSubMenuHit)
+						{
+							DelayCloseSubMenu();
+							args.Handled = true;
+						}
+					}
+				}
+
+				UpdateOwnerVisualState();
+			}
+		}
+
+		// PointerPressed event handler ensures that we're in the pressed state.
+		internal void OnPointerPressed(PointerRoutedEventArgs args)
+		{
+#if CMH_DEBUG
+    (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerPressed.", this));
+#endif // CMH_DEBUG
+
+			m_isPressed = true;
+
+			args.Handled = true;
+		}
+
+		// PointerReleased event handler shows the sub menu in the
+		// case of touch input.
+		internal void OnPointerReleased(PointerRoutedEventArgs args)
+		{
+			Pointer pointer;
+			PointerDeviceType pointerDeviceType;
+
+			m_isPressed = false;
+
+			pointer = args.Pointer;
+			pointerDeviceType = pointer.PointerDeviceType;
+
+#if CMH_DEBUG
+    (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPointerReleased - pointerDeviceType=%d.", this, pointerDeviceType));
+#endif // CMH_DEBUG
+
+			// Show the sub menu in the case of touch input.
+			// In case of the mouse device or pen, the sub menu will be shown whenever the pointer is over
+			// the sub menu owner.
+			if (PointerDeviceType.Touch == pointerDeviceType)
+			{
+				OpenSubMenu();
+			}
+
+		   args.Handled = true;
+		}
+
+		internal void OnGotFocus(RoutedEventArgs args)
+		{
+#if CMH_DEBUG
+    (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnGotFocus.", this));
+#endif // CMH_DEBUG
+
+			UpdateOwnerVisualState();
+		}
+
+		internal void OnLostFocus(RoutedEventArgs args)
+		{
+#if CMH_DEBUG
+    (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnLostFocus.", this));
+#endif // CMH_DEBUG
+
+			m_isPressed = false;
+			UpdateOwnerVisualState();
+		}
+
+		// KeyDown event handler that handles the keyboard navigation between
+		// the menu items and shows the sub menu in the case where we hit
+		// the enter, space, or right arrow keys.
+		internal void OnKeyDown(KeyRoutedEventArgs args)
+		{
+			bool handled = false;
+
+			handled = (args.Handled);
+
+#if CMH_DEBUG
+    (DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnKeyDown - handled=%d.", this, handled));
+#endif // CMH_DEBUG
+
+			if (!handled)
+			{
+				var key = VirtualKey.None;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'key'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)